astropy/units/function/logarithmic.py

```diff
@@ -64,6 +61,7 @@ def from_physical(self, x):
         """Transformation from value in physical to value in logarithmic units.
         Used in equivalency.
         """
+        # dex imported at bottom to avoid circular dependency
```

Comment:
Maybe it is actually cleaner to just do a local import here and in `to_physical` below. Happy either way.

Reply:
Hm. IDK. I agree it's cleaner to do a local import, but the single import is also nice in that it's only done once.

It feels like there's a way around this, like differently breaking up the files. E.g. a `decibel.py` file with both `DecibelUnit` and `dB` ?

Reply:
Don't know about splitting this up. Things really belong together here. In terms of imports, my original organization was more logical, as it did not needed any local imports. But having the units defined in one place is also really nice...

Reply:
I looked a bit more at reorganizing, but couldn't get it to work. In the end, I decided to go with local imports for the two cases that this not yet have them, so it looks a bit nicer. In the end, the moment one takes a log or exponential, the import time is not that important any more...

Reply:
What was the issue? I'm just eyeballing it here and it looks like

```
logarithmic.py <- LogUnit, DexUnit, dex, LogQuantity, Dex
magnitude.py <- MagUnit, mag, Magnitude
decibel.py <- DecibelUnit, dB, Decibel
units.py <- collect units here for nice summary.
```

Reply:
I'm also happy to move this discussion to a followup Issue.

Reply:
Indeed, a simpler alternative has me simply importing `STmag` etc inside `.units` (from `logarithmic`, where they were defined before). I mostly wanted to avoid making the place where units are exposed as the place of implementation be different from where they are actually defined, i.e., I tried to make this more like `si`, `cgs` and `astrophys`.

But maybe we should indeed just let it be like this for now, which (hopefully) solves the double-linking issue for the various classes and functions. The whole module could probably use a bit of thought about its organization...